client/src/components/IndicatorsDatasets.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { I18n } from "@lingui/core";
+import { t, Trans, plural } from "@lingui/macro";
+import { withI18n } from "@lingui/react";
+
+export type IndicatorsDatasetId = 'complaints'|'viols'|'permits';
+
+/**
+ * This interface encapsulates metadata about an Indicators dataset.
+ */
+export interface IndicatorsDataset {
+  /** The localized name of the dataset, e.g. "HPD Complaints". */
+  name: (i18n: I18n) => string,
+
+  /**
+   * The name to use for the dataset in analytics. This defaults to the dataset ID but
+   * can be overridden by defining this property. */
+  analyticsName?: string,
+
+  /**
+   * The localized name for a particular "quantity" of the dataset, e.g.
+   * "15 HPD Complaints issued since 2014".
+   */
+  quantity: (i18n: I18n, value: number) => string,
+
+  /**
+   * A localized explanation for what the dataset means, and where to find more information.
+   */
+  explanation: (i18n: I18n) => JSX.Element,
+}
```

Comment:
@sraby I factored out a new module, `IndicatorsDatasets`, which abstracts out the concept of a "dataset" to make the indicators visualization code a bit more DRY and less branchy.  Right now it mostly only deals with localization-related data/functionality for datasets but it could be expanded to provide more in the future.

Reply:
Ok cool! I should've warned you about a WIP branch I have called [timeline-page-v2-updates](https://github.com/JustFixNYC/who-owns-what/tree/timeline-page-v2-updates) that also abstracted out the concept of a "dataset" wherever possible in the Indicators code. Im sure when the time comes we can figure out the most strategic way to merge